system_metrics_collector/test/system_metrics_collector/test_linux_memory_measurement.cpp

```diff
@@ -143,37 +227,210 @@ class LinuxMemoryMeasurementTestFixture : public ::testing::Test
-}
+    // setting expected_stats[0]
+    moving_average_statistics::MovingAverageStatistics stats_calc;
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[0]));
+    StatisticData data = stats_calc.getStatistics();
+    StatisticDataToStatisticDataPoints(data, expected_stats[0]);
+
+    // setting expected_stats[1]
+    stats_calc.reset();
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[1]));
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[2]));
+    data = stats_calc.getStatistics();
+    StatisticDataToStatisticDataPoints(data, expected_stats[1]);
+
+    // setting expected_stats[2]
+    stats_calc.reset();
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[3]));
+    data = stats_calc.getStatistics();
+    StatisticDataToStatisticDataPoints(data, expected_stats[2]);
+
+    // setting expected_stats[3]
+    stats_calc.reset();
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[5]));
+    data = stats_calc.getStatistics();
+    StatisticDataToStatisticDataPoints(data, expected_stats[3]);
+
+    // setting expected_stats[4]
+    stats_calc.reset();
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[6]));
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[7]));
+    data = stats_calc.getStatistics();
+    StatisticDataToStatisticDataPoints(data, expected_stats[4]);
+
+    // setting expected_stats[5]
+    stats_calc.reset();
+    stats_calc.addMeasurement(processMemInfoLines(SAMPLES[8]));
+    data = stats_calc.getStatistics();
+    StatisticDataToStatisticDataPoints(data, expected_stats[5]);
+  }
+
+  int getNumReceived() const
+  {
+    return times_received;
+  }
+
+private:
+  void MetricsMessageCallback(const MetricsMessage & msg)
+  {
+    ASSERT_GT(expected_stats.size(), times_received);
+
+    // check source names
+    EXPECT_EQ(TEST_NODE_NAME, msg.measurement_source_name);
+    EXPECT_EQ("memory_usage", msg.metrics_source);
+
+    // check measurement window
+    // std::chrono::seconds window_sec(msg.window_stop.sec - msg.window_start.sec);
+    // std::chrono::nanoseconds window_nanosec(msg.window_stop.nanosec - msg.window_start.nanosec);
+    // std::chrono::milliseconds window = std::chrono::duration_cast<std::chrono::milliseconds>(
+    //   window_sec + window_nanosec);
+    // EXPECT_GT(5, std::abs(window.count() - PUBLISH_PERIOD.count()));
+
+    // check measurements
+    for (int i = 0; i < STATISTICS_DATA_TYPES.size(); ++i) {
+      EXPECT_EQ(STATISTICS_DATA_TYPES[i], msg.statistics[i].data_type);
+      if (std::isnan(expected_stats[times_received][i].data)) {
+        EXPECT_TRUE(std::isnan(msg.statistics[i].data));
+      } else {
+        EXPECT_DOUBLE_EQ(expected_stats[times_received][i].data, msg.statistics[i].data);
+      }
+    }
+
+    ++times_received;
+  }
+
+  rclcpp::Subscription<MetricsMessage>::SharedPtr subscription;
+  std::array<StatisticDataPoint[STATISTICS_DATA_TYPES.size()], 6> expected_stats;
```

Comment:
Could we clean up that a bit?

Reply:
`expected_stats` is no longer a `std::array` of C-arrays. It is now a `std::array` of `std::unordered_map`s.